### PR TITLE
added hindi and english language to the instrustion and inputform

### LINKED ISSUE #157: [C4GT Community]: Multi-language compatibility
## Description
 add an option to pick languages.

## Goals
- [ ] Instructions in Hindi
- [ ] Other details in Hindi
- [ ] Ideas to extend to other languages

### Expected Outcome

Frontend displays text in both Hindi and English. 

### Acceptance Criteria

_No response_

### Implementation Details

See how Portal Frontend (https://github.com/avantifellows/portal-frontend/) has implemented support for Hindi

### Mockups/Wireframes

_No response_

### Product Name

Quiz Frontend

### Organisation Name

Avanti Fellows

### Domain

Education

### Tech Skills Needed

Vue.js

### Mentor(s)

@suryabulusu 

### Complexity

High

### Category

Frontend

### PR DESCRIPTION
**Pull request by ananyajain10**
Fixes #157 
image

image

image

## Summary
The method changes the application's locale dynamically using Vue's $i18n instance based on the selected language.
This change allows users to dynamically switch the language of the instruction page, improving the application's multilingual support.

## Test Plan

- [ ] Test Responsiveness
  - [ ] Laptop (1200px)
  - [ ] Tablet (760px)
  - [ ] Phone (320px)
- [ ] Cross-Browser Testing
  - [ ] Chrome
  - [ ] Firefox
- [ ] ~Local Language Support~
- [ ] Hygiene and Housekeeping
  - [ ] Self-review
  - [ ] Comments have been added appropriately
  - [ ] Check for bundle size [here](https://bundlephobia.com/) if adding a package
  - [ ] Added relevant details like Labels/Projects/Milestones etc.
  - [ ] If adding or removing any environment variable, update `docs/ENV.md`, `.env.example` and the Github workflows.
- [ ] Testing
  - [ ] Wrote tests
  - [ ] Tested locally
  - [ ] Tested on staging
  - [ ] Tested on an actual physical phone
  - [ ] Tested on production
- [ ] Lighthouse Checks
  - [ ] Images have `alt` attributes
  - [ ] Any `` tags have `width` and `height` specified
  - [ ] Any `target="_blank"` links have `rel="noopener"`
  - [ ] Only SVGs are used as images. If PNGs are used, their size has been optimised.
  - [ ] Any SVG buttons without text have their `aria-label` attributes set

### CODE REVIEW

**Comment by VedantKhairnar** [comment]
Hello @suryabulusu 
Will you pls review this PR?
Thanks.

File: src/views/FormResults.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-    <div class="h-full">
+    <div class="h-full">z
```
Comment (suryabulusu) [inline src/views/FormResults.vue:2]: remove the letter

File: src/main.ts [hunk 2]
```diff
@@ -27,6 +42,8 @@ const filterBeforeCreate = (
   return toast;
 };
 
+app.use(store);
```
Comment (suryabulusu) [inline src/main.ts:45]: whats this line for

File: src/locales/en.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
+{
```
Comment (suryabulusu) [inline src/locales/en.json:1]: Sorry, but these are not the instructions i was looking for. i should have referred to `InstructionPage.vue` specifically in the Issue. 

We do not use FormInput or FormResults in Quiz Engine anymore.

File: src/locales/hi.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
+{
```
Comment (suryabulusu) [inline src/locales/hi.json:1]: Sorry, but these are not the instructions i was looking for. i should have referred to InstructionPage.vue specifically in the Issue.

We do not use FormInput or FormResults in Quiz Engine anymore.

**Review by suryabulusu — COMMENTED** [review]
@ananyajain10 Sorry, but these are not the instructions i was looking for. i should have referred to InstructionPage.vue specifically in the Issue.

We do not use FormInput or FormResults in Quiz Engine anymore.

But the structure seems mostly correct so it shouldn't take long to make these changes for InstructionPage.vue

Another suggestion: locales folder can have more meaningful file names instead of just en.json and hi.json

File: src/locales/hi.json (hunk 4 above)
Comment (ananyajain10) [reply src/locales/hi.json:1]: THANK YOU for the review. I am working towards the required changes. 
Sorry for the confusion. 
Just to clarify that only "General Instructions" part need the required changes or 
" Answering a Question: "  also?
Thank You
image

File: src/views/FormResults.vue (hunk 1 above)
Comment (ananyajain10) [reply src/views/FormResults.vue:2]: Done

**Comment by ananyajain10** [comment]
> @ananyajain10 Sorry, but these are not the instructions i was looking for. i should have referred to InstructionPage.vue specifically in the Issue.
> 
> We do not use FormInput or FormResults in Quiz Engine anymore.
> 
> But the structure seems mostly correct so it shouldn't take long to make these changes for InstructionPage.vue
> 
> Another suggestion: locales folder can have more meaningful file names instead of just en.json and hi.json


image
image

General Instructions section multilingual support for InstructionPage.vue
Please check if the required changes are met.
Thank You

**Comment by suryabulusu** [comment]
It seems you have not pushed any of the changes @ananyajain10

**Comment by ananyajain10** [comment]
> It seems you have not pushed any of the changes @ananyajain10

I was waiting for you to let me know if these are the required changes.

**Comment by ananyajain10** [comment]
@suryabulusu 
Required changes are done. Please review.
image
image

File: src/router/index.ts [hunk 5]
```diff
@@ -21,6 +21,15 @@ const routes = [
       requiresAuth: false,
     },
   },
+  {
+    path: "/InstructionPage",
```
Comment (suryabulusu) [inline src/router/index.ts:25]: no -- we don't want to provide this endpoint. i dont think this is required in your code, too? correct?

File: src/views/FormInput.vue [hunk 6]
```diff
@@ -1,259 +1,246 @@
 <template>
```
Comment (suryabulusu) [inline src/views/FormInput.vue:1]: Can you reset this file to its original form?

**Comment by suryabulusu** [comment]
@ananyajain10 left more comments and made some changes. can you take a pull of the changes i made and fix few things? we can merge then.

**Review by suryabulusu — APPROVED** [review]
looks great. made some more changes. merging soon. thanks @ananyajain10!

**Pull request merged**